pkgs/applications/misc/signumone-ks/default.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, fetchurl, dpkg, autoPatchelfHook, makeWrapper,
+  atk, ffmpeg, ffmpeg_2, ffmpeg_3, gdk-pixbuf, glibc, gtk2-x11, gtk3, libav_0_8, 
+  libXtst, pango }:
+
+stdenv.mkDerivation rec {
+  pname = "signumone-ks";
+  version = "3.1.2";
+
+  src = fetchurl {
+    url = "https://cdn-dist.signum.one/${version}/${pname}-${version}.deb";
+    sha256 = "4efd80e61619ccf26df1292194fcec68eb14d77dfcf0a1a673da4cf5bf41f4b7";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = [
+    atk glibc gdk-pixbuf stdenv.cc.cc ffmpeg ffmpeg_2 ffmpeg_3
+    libav_0_8 gtk2-x11 gtk3 pango libXtst
+  ];
+
+  libPath = stdenv.lib.makeLibraryPath buildInputs;
+
+  unpackPhase = ''
+    dpkg-deb -x ${src} ./
+  '';
+
+  installPhase = ''
+    DESKTOP_PATH=$out/share/applications/signumone-ks.desktop
+    LIB_DIR=$out/lib/signumone-ks
```

Comment:
You should not have to deal with this. Try just putting `ffmpeg_2` into `buildInputs`.

Reply:
If I don't add `ffmpeg` nor `ffmpeg_3`, I get missing libraries on the build process:
```
these derivations will be built:
  /nix/store/56pva1pz92dpgqbifgp1h54740dymkcd-signumone-ks-3.1.2.drv
building '/nix/store/56pva1pz92dpgqbifgp1h54740dymkcd-signumone-ks-3.1.2.drv'...
unpacking sources
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
glibPreInstallPhase
installing
glibPreFixupPhase
dropIconThemeCache
dropIconThemeCache
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/SignumOne-KS
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libfxplugins.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libsunec.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libinstrument.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglass.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglib-lite.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjsdt.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/liblcms.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/server/libjvm.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/server/libjsig.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libawt_headless.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libprism_common.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjavafx_font_pango.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libnio.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libj2pkcs11.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjfr.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-56.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libsctp.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libdecora_sse.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-54.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libawt_xawt.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjavafx_iio.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libmanagement.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-58.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libresource.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjdwp.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libgstreamer-lite.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjfxmedia.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libhprof.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libprism_sw.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjsound.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/jli/libjli.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libnpt.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjpeg.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libsplashscreen.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglassgtk3.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjaas_unix.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjfxwebkit.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglassgtk2.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libattach.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjavafx_font_freetype.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjava_crw_demo.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libdt_socket.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjsoundalsa.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libnet.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libawt.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libj2pcsc.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjava.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libj2gss.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-57.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libt2k.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libsaproc.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libverify.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libprism_es2.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjavafx_font.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libfontmanager.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjsig.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libunpack.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libdcpr.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-55.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libdeploy.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-53.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libbci.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libjawt.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libzip.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libmlib_image.so
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/jexec
shrinking /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/libpackager.so
strip is /nix/store/hiwz81i1g3fn3p0acjs042a4h5fri6dh-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/lib  /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/bin 
patching script interpreter paths in /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2
checking for references to /build/ in /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2...
automatically fixing dependencies for ELF files
searching for dependencies of /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/SignumOne-KS
  libstdc++.so.6 -> found: /nix/store/fym51l26jrpzhlvaj607wd3im9511j2q-gcc-9.3.0-lib/lib/libstdc++.so.6
setting RPATH to: /nix/store/fym51l26jrpzhlvaj607wd3im9511j2q-gcc-9.3.0-lib/lib
searching for dependencies of /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libfxplugins.so
  libgstreamer-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libgstreamer-lite.so
  libglib-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglib-lite.so
setting RPATH to: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64:/nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64
searching for dependencies of /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-56.so
  libgstreamer-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libgstreamer-lite.so
  libglib-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglib-lite.so
  libavcodec-ffmpeg.so.56 -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/lib/signumone-ks/libavcodec-ffmpeg.so.56
  libavformat-ffmpeg.so.56 -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/lib/signumone-ks/libavformat-ffmpeg.so.56
setting RPATH to: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64:/nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64:/nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/lib/signumone-ks:/nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/lib/signumone-ks
searching for dependencies of /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libsunec.so
searching for dependencies of /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libavplugin-ffmpeg-57.so
  libgstreamer-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libgstreamer-lite.so
  libglib-lite.so -> found: /nix/store/10wkm930nii0680yw3ch3dyvxh8i2vkk-signumone-ks-3.1.2/opt/SignumOne-KS/runtime/lib/amd64/libglib-lite.so
  libavcodec.so.57 -> not found!
  libavformat.so.57 -> not found!
builder for '/nix/store/56pva1pz92dpgqbifgp1h54740dymkcd-signumone-ks-3.1.2.drv' failed with exit code 1
error: build of '/nix/store/56pva1pz92dpgqbifgp1h54740dymkcd-signumone-ks-3.1.2.drv' failed
```